### PR TITLE
Make tbtacl safer to use with untrusted input

### PR DESCRIPTION
**Pull request by benley**
This is mostly just the result of running shellcheck and following its advice.

### CODE REVIEW

File: tbtacl/tbtacl.in [hunk 1]
```diff
@@ -114,24 +114,24 @@ esac
...
 case "$action" in
 	# New device attached, go to authorize it
-	add)	authorize $device
+	add)	authorize "$device"
 		;;
 
 	# The device got authorized, let's try to authorize again the devices
 	# behind it
-	change)	list=$device/*/authorized
+	change)	list="$device/*/authorized"  # this glob is expanded later
```
Comment (YehezkelShB) [inline tbtacl/tbtacl.in:122]: later == line 124?

**Review by YehezkelShB — COMMENTED** [review]
Thanks for stepping in!

Most changed variables can never include a space, but I don't see any reason not to quote them.

File: tbtacl/tbtacl.in (hunk 1 above)
Comment (benley) [reply tbtacl/tbtacl.in:122]: Line 126, in `for i in $list ; do`

Comment (benley) [reply tbtacl/tbtacl.in:122]: It already was not being expanded on line 122 btw - globs and brace expansions are literals in assignments.  Adding quotes just makes it more explicit to the reader.

Comment (YehezkelShB) [reply tbtacl/tbtacl.in:122]: It must be expanded in line 124 too, otherwise it'll fail. Does it?
(I have no way to test it right now. Did you?)

Comment (benley) [reply tbtacl/tbtacl.in:122]: Line 124 just needs to check if the `*` character appears in the $list string, so it should still be OK.  Probably could drop the `-F` from grep as well - or I could remove the quotes around $list here if you prefer.

It does seem to work this way on my system.

Comment (benley) [reply tbtacl/tbtacl.in:122]: oh crap - I was completely incorrect in my last comment. 124 does need to expand the glob in order for the test to work.  Fixing.

Comment (benley) [reply tbtacl/tbtacl.in:122]: (fixed)

**Review by YehezkelShB — APPROVED** [review]
LGTM. Thanks!

@finikorg, do you want to take a look too (or just merge it)?

**Pull request merged**